zerver/lib/actions.py

```diff
@@ -2609,6 +2609,23 @@ def truncate_topic(topic):
     # type: (Text) -> Text
     return truncate_content(topic, MAX_SUBJECT_LENGTH, "...")
 
+def do_update_buddy_list(user_profile, buddy_profile, should_add):
+    # type: (UserProfile, str, bool) -> None
+
+    if should_add:
+        BuddyList.objects.get_or_create(user=user_profile, buddy=buddy_profile)
+    else:
+        BuddyList.objects.filter(user=user_profile, buddy=buddy_profile).delete()
+
+def get_buddy_list(user_profile):
+    # type: (UserProfile) -> List[int]
+
+    records = BuddyList.objects.filter(user=user_profile)
+    user_list = []
+    for record in records:
+        user_list.append(record.buddy.id)
```

Comment:
Actually, even better would be:

~~~ py
user_id_set = {r.buddy.id. for r in record}
return user_id_set